Add common-case visibility scopes to `Status`

mjankowski opened the pull request:
There are various spots around the app which use these specific combinations of visibility. This adds scopes to capture that.

Naming-wise, I chose `distributable_` based on an existing query method with that name for that value combination. I chose `home_feed_` based on some of the FeedManager usage. Assuming the idea in general is fine, open to naming feedback on either one.

mjankowski committed:
Add visibility scopes to Status

ClearlyClaire commented:
I'm not sure about `home_feed_visibility`, as posts with *any* visibility can appear in the home feed (though I suppose this highlights a minor edge case, where Private Mentions do not get back-fitted into the home timeline when following a person you previously exchanged with).

mjankowski commented:
Yeah, it sorta makes sense in the pr given the limited number of uses, but I agree overall on naming. Any better ideas?

mjankowski commented:
Rebased and renamed `home_feed_visibility` scope to `list_eligible_visibility`. Seems slightly more accurate (or less potentially inaccurate) given usage in `FeedManager`. Open to better ideas.

mjankowski commented:
Closing in favor of linked extracted smaller PRs.
